frontend/src/scenes/dashboard/dashboardLogic.ts

```diff
@@ -126,28 +121,78 @@ export const dashboardLogic = kea<dashboardLogicType>({
                         actions.setDates(dashboard.filters.date_from, dashboard.filters.date_to, false)
                         return dashboard
                     } catch (error: any) {
-                        actions.setReceivedErrorsFromAPI(true)
                         if (error.status === 404) {
                             return null
                         }
                         throw error
                     }
                 },
+                updateLayouts: async ({ layouts }) => {
+                    if (!isUserLoggedIn()) {
+                        // If user is anonymous (i.e. viewing a shared dashboard logged out), we don't save any layout changes.
+                        return
+                    }
+                    if (!props.id) {
+                        // what are we saving layouts against?!
+                        return
+                    }
+
+                    const itemLayouts: Record<number, Partial<Record<string, Layout>>> = {}
+
+                    Object.entries(layouts).forEach(([col, layout]) => {
+                        layout.forEach((layoutItem) => {
+                            // it _should_ be impossible for a layout item to not be in this mapping
+                            const insightId = values.shortIdToId?.[layoutItem.i]
+                            if (insightId) {
```

Comment:
Random thought: maybe `if (!insightId) { throw }` to make this explicit? 🤷 
No strong feelings either way